explorer/interpreter/type_checker.cpp

```diff
@@ -1700,28 +1700,28 @@ class TypeChecker::ConstraintTypeBuilder {
     friend class ConstraintTypeBuilder;
 
    private:
-    int num_impls_added = 0;
-    int num_equals_added = 0;
+    int num_impl_constraints_added = 0;
```

Comment:
This name is a bit confusing because it says `impl` but `constraints`.

But I think this actually points to a whole section of code that might need to be updated, although maybe only minorly? (I also can't drop a comment on those lines, yay.)

Above, there is an `AddImplConstraint` method on the builder. The comments for that seem like they need to be updated (`T is C` and such), but it also isn't completely clear what terminology should be used. On one hand, it is used to track the new `impls` where clause. But it is operating an an `ImplConstraint` type with variables named `impl` etc. My guess is that this section may just need a more general update to pick a consistent set of terminology here and decide whether `ImplConstraint` or something else makes the most sense.

Anyways, this variable name is clearly correct until and unless the rest of the terminology shifts. I'd guess that even the comment updates might make sense as a follow-up anyways to at least fix lingering uses of `is` and if desired adjust terminology or add any explanation of terminology that is useful.

Reply:
If I understand correctly, you are asking me to replace "impl constraint" with "impls constraint", which I've now done, along with three instances of `T is C` with `T impls C` in comments.